keras/backend/tensorflow_backend.py

```diff
@@ -399,7 +399,7 @@ def gradients(loss, variables):
 # CONTROL FLOW
 
 def rnn(step_function, inputs, initial_states,
-        go_backwards=False, mask=None):
+        go_backwards=False, mask=None, constants=None):
```

Comment:
I see no need to add a `constants` argument to the `rnn` API. I can be made part of the states (and the whole logic will be handled at the level of the `step` function of the RNN layers, as it should be...).